x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/source_settings.tsx

```diff
@@ -184,7 +186,7 @@ export const SourceSettings: React.FC = () => {
           href={diagnosticsPath}
           isLoading={buttonLoading}
           data-test-subj="DownloadDiagnosticsButton"
-          download
+          download={`${id}_${serviceType}_${Date.now()}_diagnostics.json`}
```

Comment:
For reference: https://github.com/elastic/ent-search/blob/master/frito_pie/app/controllers/frito_pie/content_sources_controller.rb#L197

I'm basically just mimicking in native HTML5 the filename that the server sends back. Date timestamps are slightly different due to JS vs Ruby, but likely not an issue